src/data.js

```diff
@@ -263,45 +263,154 @@ let Artifact = Entity.configure({
+  },
+  context: [
+    'blobStore',
+    'privateBucket',
+    'publicBucket',
+    'monitor',
+  ],
+});
+
+/** Entity for tracking artifacts.
+ * Version 2 adds 'present' field */
+let Artifact = Entity.configure({
+  version: 2,
+  properties: {
+    taskId: Entity.types.SlugId,
+    runId: Entity.types.Number,
+    name: Entity.types.String,
+    storageType: Entity.types.String,
+    contentType: Entity.types.String,
     /**
      * Location details storageType,
      *
-     * storageType: s3
-     *   bucket:        S3 bucket that contains the object
-     *   prefix:        Prefix (path) for the object within the bucket
-     *
-     * storageType: azure
-     *   container:     Azure container that holds the blob
-     *   path:          Path to blob within container
+     * storageType: blob
+     *   provider: which provider stores this artifact (e.g. s3)
+     *   region: which region this artifact is in (e.g. us-west-2)
+     *   bucket: name of the 'folder' the artifact is stored in (e.g.
+     *     taskcluster-public-artifacts)
+     *   key: name of the 'file' in the storage (e.g. $taskid/$runid/$name)
+     *   length: As per the HTTP Content-Length header
+     *   md5: hex encoded MD5 digest of content
+     *   sha256: hex encoded SHA256 digest of content
+     * NOTE: the bucket/key terminology is s3 terminology, but other blob
+     * stores have conceptually equivalent notions.  We chose to use s3
+     * terminology because that is widely known and so far our only useful
+     * implementation of the blob storage type.  In azure, these are containers
+     * and blobs, respectively.  In google cloud storage, these are bucket and
+     * object, respectively.
      *
      * storageType: reference
      *   url:           URL that artifact should redirect to
      *
      * storageType: error
      *   reason:        Formalized reason for error artifact, see JSON schema.
      *   message:       Human readable error message to return
+     *
+     * storageType: s3 (deprecated)
+     *   bucket:        S3 bucket that contains the object
+     *   prefix:        Prefix (path) for the object within the bucket
+     *
+     * storageType: azure (deprecated)
+     *   container:     Azure container that holds the blob
+     *   path:          Path to blob within container
+     *
      */
-    details:        Entity.types.JSON,
-    expires:        Entity.types.Date,
+    details: Entity.types.JSON,
+    expires: Entity.types.Date,
+    present: Entity.types.Number, // 0 or 1 to represent whether the artifact is there
+  },
+  migrate: function (item) {
+    // Add the present flag.  We will set this value to 1 for all cases because
+    // they are either stored in this table and can be worked on atomically so
+    // are always present if the entity can be loaded or because they are the
+    // deprecated s3 or azure storageType which will never have checking for
+    // presence
+    return {
```

Comment:
In the past you said that I shouldn't do that, which is why I created a new object.  I'll happily do that